### DISCUSSION TITLE
Linux AppImage Fails on Debian 12 bookworm

### DISCUSSION THREAD

**Discussion by tuxxxed** [OrcaSlicer/OrcaSlicer#5448] [Q&A]
Hi All,

Trying to use the OrcaSlicer_Linux_V2.0.0.AppImage .

My Linux system is as follows:
Base: Debian GNU/Linux 12 (bookworm)
Distro: antiX-23-runit_x64-base
CPU: Intel Core i7-6700
Graphics: Intel HD Graphics 530
System Memory: 32 GiB
Shell: Bash 5.2.15
Desktop: JWM

Have downloaded the AppImage and used chmod+x to enable running from the command line.

When launching, many errors ensue ending in a segmentation fault.

Terminal output as follows:

$ ./OrcaSlicer_Linux_V2.0.0.AppImage)
[2024-05-22 18:21:05.095581] [0x00007fce93891d40] [trace]   Initializing StaticPrintConfigs
add font of HarmonyOS_Sans_SC_Bold returns 1
add font of HarmonyOS_Sans_SC_Regular returns 1
add font of NanumGothic-Regular returns 1  
add font of NanumGothic-Bold returns 1 

(orca-slicer:14614): Gtk-CRITICAL **: 18:21:05.463: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkScrollbar

(orca-slicer:14614): Gtk-CRITICAL **: 18:21:10.743: gtk_window_resize: assertion 'width > 0' failed

(orca-slicer:14614): GLib-GObject-WARNING **: 18:21:10.765: invalid cast from 'wxPizza' to 'GtkCellLayout'

(orca-slicer:14614): Gtk-CRITICAL **: 18:21:10.765: gtk_cell_layout_get_cells: assertion 'GTK_IS_CELL_LAYOUT (cell_layout)' failed

(orca-slicer:14614): Gtk-CRITICAL **: 18:21:11.460: gtk_window_resize: assertion 'height > 0' failed

(orca-slicer:14614): Gtk-CRITICAL **: 18:21:11.464: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:14614): Gtk-CRITICAL **: 18:21:11.472: gtk_window_resize: assertion 'height > 0' failed

(orca-slicer:14614): Gtk-CRITICAL **: 18:21:11.473: gtk_widget_set_size_request: assertion 'width >= -1' failed

(orca-slicer:14614): Gtk-CRITICAL **: 18:21:11.598: gtk_window_resize: assertion 'width > 0' failed

(orca-slicer:14614): Gtk-CRITICAL **: 18:21:11.606: gtk_widget_set_size_request: assertion 'width >= -1' failed
segmentation fault
$

Does anyone have an idea why the AppImage does not work with a mainstream Linux distribution?
Does anyone have suggestions on how to troubleshoot these problems?

Look forward to the feedback.

**Comment by ronoaldo** [comment]
Hi! Try these steps to diagnose the issue. It could be that a missing library that is not bundled in the AppImage is causing this.

First, unpack the .AppImage with:

    orcaslicer.appimage --appimage-extract # replace with the name of the file you downloaded

Then, a folder named squashfs-root will be unpacked. This command may show you which libraries are missing, if any:

    LANG=C ldd squashfs-root/bin/orca-slicer  | grep -i 'not found'

Install the packages and you should be good to go. I'm running it on Debian 12 KDE, using the V2.0.0 released .AppImage, but I had to do the previous procedure in order to have it working for me.

**Reply by tuxxxed** [reply]
Hi ronoaldo,

Thank you for the reply. I executed the suggestions as instructed.

In particular the grep command:

LANG=C ldd squashfs-root/bin/orca-slicer  | grep -i 'not found'

Nothing missing was found in squashfs-root/bin/orca-slicer.

So, this suggests that there are no missing libraries.

I suspect that the orca-slicer binary file within the AppImage was compiled with an option that is not compatible with my Linux distribution.

If you have further suggestions, please let me know.

**Reply by ronoaldo** [reply]
I did some testing and it looks like there are a few options you may want to try.

I managed to trigger a segmentation fault by running the AppImage with '--help', so there seems to be a problem indeed with the AppImage. Without this option, it worked out for me and the program launches normaly. I was able to pass an invalid option, such as '-a' and it prints the help so I really don't know what is going on!

I also managed to trigger an error by using the 'orcaslicer.appimage --appimage-portable-home' option first, then running the program. This option creates a "portable home folder" to act as the $HOME for the app image. I figured it does not creates the folder ~/.config, only the ~/.config/OrcaSlicer subfolder. I managed to launch with portable home after creating the folder manually in the portable home. This can also be useful to test configs/upgrades without messing with your profiles.

You can also try to install 'strace' and at least have some tracing before the crash. To me, it crashes after checking for a file when I run --help:

```
ATX_BASIC_STATS|STATX_MNT_ID, stx_attributes=0, stx_mode=S_IFDIR|0755, stx_size=4096, ...}) = 0
statx(AT_FDCWD, "/home/ronoaldo/Applications/squashfs-root", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW|AT_NO_AUTOMOUNT, STATX_TYPE|STATX_MODE, {stx_mask=STATX_BASIC_STATS|STATX_MNT_ID, stx_attributes=0, stx_mode=S_IFDIR|0755, stx_size=4096, ...}) = 0
statx(AT_FDCWD, "/home/ronoaldo/Applications/squashfs-root/bin", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW|AT_NO_AUTOMOUNT, STATX_TYPE|STATX_MODE, {stx_mask=STATX_BASIC_STATS|STATX_MNT_ID, stx_attributes=0, stx_mode=S_IFDIR|0755, stx_size=4096, ...}) = 0
statx(AT_FDCWD, "/home/ronoaldo/Applications/squashfs-root/bin/orca-slicer", AT_STATX_SYNC_AS_STAT|AT_SYMLINK_NOFOLLOW|AT_NO_AUTOMOUNT, STATX_TYPE|STATX_MODE, {stx_mask=STATX_BASIC_STATS|STATX_MNT_ID, stx_attributes=0, stx_mode=S_IFREG|0755, stx_size=113179048, ...}) = 0
brk(0x56160d40e000)                     = 0x56160d40e000
stat("", 0x7ffe82bc79a0)                = -1 ENOENT (Arquivo ou diretório inexistente)
stat("", 0x7ffe82bc79a0)                = -1 ENOENT (Arquivo ou diretório inexistente)
stat("", 0x7ffe82bc79a0)                = -1 ENOENT (Arquivo ou diretório inexistente)
stat("/tmp", {st_mode=S_IFDIR|S_ISVTX|0777, st_size=163840, ...}) = 0
newfstatat(AT_FDCWD, "/etc/localtime", {st_mode=S_IFREG|0644, st_size=1444, ...}, 0) = 0
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0x8} ---
+++ killed by SIGSEGV (core dumped) +++
Falha de segmentação (imagem do núcleo gravada)
```

If everything fails, it could be worth opening a bug report with the strace output and if you managed to grab it, the core dump file as well.
